benchmark848/benchmark.cpp

```diff
@@ -0,0 +1,348 @@
+#include <iostream>
+#include <string>
+#include "build/debug/src/kuzu.hpp"
+#include <algorithm>
+
+using namespace kuzu::main;
+using namespace std;
+
+/* We create three node tables: People, Customer, and Organization. All three tables contain 10k entries */
+const int NUM_TABLES = 3;
+const int NUM_ROWS = 10000;
+const string tableNames[NUM_TABLES] = {"People", "Customer", "Organization"};
+const string tableCreateQuery [NUM_TABLES] = {
+    "CREATE NODE TABLE People (id INT32, firstName STRING, lastName STRING, sex STRING, email STRING, phone STRING, jobTitle STRING, PRIMARY KEY(id));",
+    "CREATE NODE TABLE Customer (id INT32, firstName STRING, lastName STRING, company STRING, city STRING, country STRING, primaryPhone STRING, secondaryPhone STRING, email STRING, website STRING, PRIMARY KEY(id));",
+    "CREATE NODE TABLE Organization (id INT32, name STRING, website STRING, country STRING, description STRING, foundYear INT16, industry STRING, numEmployee INT16, PRIMARY KEY(id));"
+};
+/* will be updated later */
+string tableCopyQuery [NUM_TABLES] = {"", "", ""};
+
+const vector<string> tableColumns[NUM_TABLES] = {{"id", "firstName", "lastName", "sex", "email", "phone", "jobTitle"},
+    {"id", "firstName", "lastName", "company", "city", "country", "primaryPhone", "secondaryPhone", "email", "website"},
+    {"id", "name", "website", "country", "description", "foundYear", "industry", "numEmployee"}};
+const int tableNumColumns[3] = {7, 10, 8};
+
+enum AlterType : int
+{
+    DROP_COLUMN = 0,
+    ADD_COLUMN = 1
+};
+
+enum AlterAddColumnType : int
+{
+    INVALID_COLUMN_TYPE = -1,
+    INT64 = 0,
+    BOOL = 1,
+    STRING = 2
+};
+
+enum TestType : int
+{
+    DROP_TABLE = 0,
+    ALTER_TABLE = 1,
+    DELETE_NODE_GROUP = 2
+};
+
+/* Helper functions */
+std::string GenRandomStr(const int len) {
+    static const char alphanum[] =
+        "0123456789"
+        "ABCDEFGHIJKLMNOPQRSTUVWXYZ"
+        "abcdefghijklmnopqrstuvwxyz";
+    std::string tmp_s;
+    tmp_s.reserve(len);
+
+    for (int i = 0; i < len; ++i) {
+        tmp_s += alphanum[rand() % (sizeof(alphanum) - 1)];
+    }
+    
+    return tmp_s;
+}
+
+char* GetCmdOption(char ** begin, char ** end, const std::string & option)
+{
+    char ** itr = std::find(begin, end, option);
+    if (itr != end && ++itr != end)
+    {
+        return *itr;
+    }
+    return nullptr;
+}
+
+void UpdateTableCopyQuery(string csvFileDir)
```

Comment:
Recommend passing strings as const references wherever possible, to avoid overhead of copying string values:
```
void updateTableCopyQuery(const string& csvFileDir);
```
Applicable throughout the file.

Reply:
Fixed in the newest commit